tooling/nargo/src/ops/foreign_calls.rs

```diff
@@ -36,16 +45,57 @@ impl ForeignCall {
             "println" => Some(ForeignCall::Println),
             "get_number_sequence" => Some(ForeignCall::Sequence),
             "get_reverse_number_sequence" => Some(ForeignCall::ReverseSequence),
+            "create_mock" => Some(ForeignCall::CreateMock),
+            "set_mock_params" => Some(ForeignCall::SetMockParams),
+            "set_mock_returns" => Some(ForeignCall::SetMockReturns),
+            "set_mock_times" => Some(ForeignCall::SetMockTimes),
+            "clear_mock" => Some(ForeignCall::ClearMock),
             _ => None,
         }
     }
+}
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+struct MockedCall {
+    id: usize,
+    name: String,
+    params: Option<Vec<ForeignCallParam>>,
+    result: ForeignCallResult,
+    times_left: Option<u64>,
```

Comment:
Could we document a bit more the purpose of `times_left`, and then also why we have a `last_mock_id` inside of the ForeignCallExecutor